crates/ibc/src/applications/transfer/amount.rs

```diff
@@ -6,10 +6,39 @@ use derive_more::{Display, From, Into};
 use super::error::TokenTransferError;
 use primitive_types::U256;
 
+#[cfg(feature = "schema")]
+use crate::alloc::{borrow::ToOwned, string::String};
+
 /// A type for representing token transfer amounts.
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Display, From, Into)]
-pub struct Amount(U256);
+pub struct Amount(
+    #[cfg_attr(feature = "schema", schemars(with = "String"))]
+    #[serde(serialize_with = "crate::serializers::serde_string::serialize")]
+    #[serde(deserialize_with = "deserialize")]
+    U256,
+);
+
+#[cfg(feature = "serde")]
+fn deserialize<'de, D>(deserializer: D) -> Result<U256, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    use serde::Deserialize;
+    U256::from_dec_str(<&str>::deserialize(deserializer)?).map_err(serde::de::Error::custom)
+}
+
+#[test]
+#[cfg(feature = "serde")]
+fn serde_amount() {
+    let value = Amount::from(42);
+    let string = serde_json::to_string(&value).expect("can serde string");
+    assert_eq!(string, "\"42\"");
+    let binary = serde_json::to_vec(&value).expect("can serde binary");
+    let de: Amount = serde_json::from_slice(binary.as_ref()).expect("can deserialize");
+    assert_eq!(de, value);
+}
```

Comment:
And move this down to the bottom of the page under `mod tests {}?` like e.g. [here](https://github.com/cosmos/ibc-rs/blob/main/crates/ibc/src/clients/ics07_tendermint/consensus_state.rs#L162-L183)